### PR TITLE
cranelift-object: reject symbol names with NUL byte without panic'ing.

### PR DESCRIPTION
**Pull request by cfallin**
Avoid a `panic!()`, and return a proper error, on a NUL byte. We hit a
null-byte check inside the `object` crate otherwise; this blocks fuzzing
when testing via a write-object-file-and-dlopen flow.

### CODE REVIEW

File: cranelift/object/src/backend.rs [hunk 1]
```diff
@@ -145,6 +145,18 @@ impl ObjectModule {
     }
 }
 
+fn validate_symbol(name: &str) -> ModuleResult<()> {
+    // null bytes are not allowed in symbol names and will cause the `object`
+    // crate to panic. Let's return a clean error instead.
+    if name.contains("\0") {
+        return Err(ModuleError::Backend(anyhow::anyhow!(
+            "Symbol '{}' has a null byte, which is disallowed",
```
Comment (alexcrichton) [inline cranelift/object/src/backend.rs:153]: It might be better to use `:?` here which will show the nul byte perhaps?

**Review by alexcrichton — APPROVED** [review]
Seems reasonable to me! If possible mind also adding a test?

File: cranelift/object/src/backend.rs (hunk 1 above)
Comment (cfallin) [reply cranelift/object/src/backend.rs]: Done.

**Comment by cfallin** [comment]
Thanks, tests added as well.

**Pull request merged**